snuba/migrations/operations.py

```diff
@@ -0,0 +1,41 @@
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+
+
+class Operation:
+    """
+    Executed on all the nodes of the cluster.
+    """
+
+    def __init__(self, storage_set: StorageSetKey):
+        self.storage_set = storage_set
+
+    def execute(self) -> None:
+        cluster = get_cluster(self.storage_set)
+
+        for node in cluster.get_local_nodes():
+            connection = cluster.get_node_connection(
+                ClickhouseClientSettings.MIGRATE, node
+            )
+            connection.execute(self.format_sql())
+
+    def format_sql(self) -> str:
+        raise NotImplementedError
```

Comment:
Does this mean all operations essentially execute one and onnly one SQL statement ? 
Is there any case where a migration would have to load data in chunks from a column, transform that and store the transformed rows into another column ?

Reply:
Good point, I will remove this from the base class.